api/src/game/game.controller.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { AppGateway } from '../app.gateway';
+import { GameState, Player } from '../types';
+import { GameService } from './game.service';
+import { INITIAL_BOARD } from '../constants';
+
+@Controller('game')
+export class GameController {
+  constructor(
+    private readonly gameService: GameService,
+    private gatewayService: AppGateway,
+  ) {}
+
+  @Post('start')
+  async createGame(): Promise<GameState> {
+    const res = await this.gameService.createGame();
+    return this.gameService.deserializer(res);
+  }
+
+  @Get(':id')
+  async getGameState(@Param('id') id: string): Promise<GameState> {
+    try {
+      const res = await this.gameService.getGame({ id: Number(id) });
+      return this.gameService.deserializer(res);
+    } catch (e) {
+      throw new HttpException("That game id doesn't exists", 400);
+    }
+  }
+
+  @Get('join/:id')
+  async joinGame(@Param('id') id: string): Promise<GameState> {
+    try {
+      const res = await this.gameService.getGame({ id: Number(id) });
+      return this.gameService.deserializer(res);
+    } catch (e) {
+      throw new HttpException("That game id doesn't exists", 400);
+    }
+  }
+
+  @Post('restart/:id')
+  async restartGame(@Param('id') id: string): Promise<GameState> {
+    if (!Number(id)) {
+      throw new HttpException('Please give a valid id', 400);
+    }
+    try {
+      const clearedPlayers = this.gameService.initializePlayers();
+      const res = await this.gameService.updateGame({
+        where: { id: Number(id) },
+        data: {
+          board: JSON.stringify(INITIAL_BOARD),
+          players: JSON.stringify(clearedPlayers),
+          currentPlayer: 1,
+        },
+      });
+      const gameState = this.gameService.deserializer(res);
+      this.gatewayService.emitGameState(gameState);
+      return gameState;
+    } catch (e) {
+      throw new HttpException("That game doesn't exists", 400);
+    }
+  }
+
+  @Put(':id')
```

Comment:
why do you need this route?

Reply:
it has been reworked and linked with the next point